numpy/core/multiarray.py

```diff
@@ -1298,9 +1305,33 @@ def shares_memory(a, b, max_work=None):
 
     Examples
     --------
-    >>> np.may_share_memory(np.array([1,2]), np.array([5,8,9]))
+    >>> x = np.array([1, 2, 3, 4])
+    >>> np.shares_memory(x, np.array([5, 6, 7]))
+    False
+    >>> np.shares_memory(x[::2], x)
+    True
+    >>> np.shares_memory(x[::2], x[1::2])
     False
 
+    Checking whether two arrays share memory is NP-complete, and
+    runtime may increase exponentially in the number of
+    dimensions. Hence, `max_work` should generally be set to a finite
+    number, as it is possible to construct examples that take
+    extremely long to run::
```

Comment:
```suggestion
    extremely long to run:
```
I think the double colon is incorrect in examples (the CI failed probably due to this).

LGTM  otherwise. I wonder if there is a simple explanation what "max_work" actually stands for (e.g. how  much is 1000 in the example). I assume there isn't really and that is fine.

Reply:
Fixed.

It says in the parameter list "maximum number of candidate solutions to consider", but I'm not sure how to elaborate on that without explaining how the whole thing works...